library/bignum_core.h

```diff
@@ -0,0 +1,125 @@
+/**
+ *  Internal bignum functions
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef MBEDTLS_BIGNUM_CORE_H
+#define MBEDTLS_BIGNUM_CORE_H
+
+#include "common.h"
+
+#if defined(MBEDTLS_BIGNUM_C)
+#include "mbedtls/bignum.h"
+#endif
+
+/** Count leading zero bits in a given integer.
+ *
+ * \param x     Integer to count leading zero bits.
+ *
+ * \return      The number of leading zero bits in \p x.
+ */
+size_t mbedtls_mpi_core_clz( const mbedtls_mpi_uint x );
+
+/** Return the number of bits of an MPI.
```

Comment:
Err, what exactly is "the number of bits of an MPI"? It's clearly not `sizeof(mbedtls_mpi_uint) * N`

Is it "the minimum number of bits required to represent the value held in the MPI"?

The doc should also make it clear that it returns `0` if all bits of the MPI are `0`.